src/js/routes/More/ProcessingDonation.jsx

```diff
@@ -23,28 +23,24 @@ export default class ProcessingDonation extends Component {
 
     this.state = {
       donation_response_received: false,
-      donation_success: false
+      donation_success: false,
     };
 
     this._onDonateStoreChange = this._onDonateStoreChange.bind(this);
   }
 
   componentDidMount () {
-//    this._onDonateStoreChange();
+    // this._onDonateStoreChange();
     this.donateStoreListener = DonateStore.addListener(this._onDonateStoreChange);
-
   }
 
-  componentWillUnmount (){
+  componentWillUnmount () {
     this.donateStoreListener.remove();
   }
 
   _onDonateStoreChange () {
-     this.setState({donation_response_received: DonateStore.donation_response_received(), donation_success: DonateStore.donation_success()});
-    }
-
-
-  render () {
+    this.setState({ donation_response_received: DonateStore.donation_response_received(),
+      donation_success: DonateStore.donation_success(), });
```

Comment:
Hi Steve, when you call this.setState, the value isn't accessible until a re-render. Moving "if (this.state.donation_response_received)" into _onDonateStoreChange will not behave the way it looks like you are seeking. It is more common to put the state-dependent reactions in the "render".